### PR TITLE
[1248] Implement ForbiddenInlineIgnoreViolation

### LINKED ISSUE #1248: Disallow to ignore some violations on line level
# Rule request

## Thesis

Sometimes developers inline ignore violations that should not be ignored. Like complexity violations, best practices, or any others that are important for this specific project.

So, we need to give users an option to provide a configuration hook with violations names (both short like `WPS` and full ones like `WPS102`) that cannot be ignored.

And when we find inline ignore of this violations - we should raise a new one on `0` line.

We do not check `per-file-ignores` and `ignored` with this check.

Related: https://github.com/wemake-services/wemake-python-styleguide/pull/1246/files#diff-2c42a0740c36d757ff4d489010b3138eR4

This is a best practices violation.

### PR DESCRIPTION
**Pull request by edytagarbarz**
# I have made things!

## Checklist

- [x] I have created at least one test case for the changes I have made
- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have updated the documentation for the changes I have made

## Related issues

Closes #1248 

🙏 Please, if you or your company is finding wemake-python-styleguide valuable, help us sustain the project by sponsoring it transparently on https://opencollective.com/wemake-python-styleguide. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

**Comment by edytagarbarz** [comment]
Hi @sobolevn I need your help here with logistic. I added this new violation  under MagicCommentVisitor but doing so I exceeded allowed number of methods. I thought about moving noqa comments checks to a separate visitor but wanted to hear your thoughts first. Thanks!

**Comment by edytagarbarz** [comment]
This is a kind reminder. @sobolevn  could you take a look please?

File: wemake_python_styleguide/visitors/tokenize/comments.py [hunk 1]
```diff
@@ -81,6 +83,29 @@ def _check_noqa(self, token: tokenize.TokenInfo) -> None:
             # We cannot pass the actual line here,
             # since it will be ignored due to `# noqa` comment:
             self.add_violation(WrongMagicCommentViolation(text=comment_text))
+            return
+        self._check_forbidden_noqa(excludes)
+
+    def _check_forbidden_noqa(self, noqa_excludes) -> None:
+        excludes_list = noqa_excludes.split(',')
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/tokenize/comments.py:90]: We can drop this intermediate variable

File: wemake_python_styleguide/visitors/tokenize/comments.py [hunk 2]
```diff
@@ -81,6 +83,29 @@ def _check_noqa(self, token: tokenize.TokenInfo) -> None:
...
+        excludes_list = [ex.strip() for ex in excludes_list]
+        forbidden_noqa = ''.join(self.options.forbidden_inline_ignore)
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/tokenize/comments.py:92]: Can we prepare this value somewhere earlier?

File: wemake_python_styleguide/visitors/tokenize/comments.py [hunk 3]
```diff
@@ -81,6 +83,29 @@ def _check_noqa(self, token: tokenize.TokenInfo) -> None:
...
+        forbidden_noqa_list = forbidden_noqa.split(',')
+        for fn in forbidden_noqa_list:
+            fn = fn.strip()
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/tokenize/comments.py:95]: What is `fn`? I think we need a better name here

File: wemake_python_styleguide/visitors/tokenize/comments.py (hunk 2 above)
Comment (edytagarbarz) [reply wemake_python_styleguide/visitors/tokenize/comments.py:91]: Sorry, I don't get the problem here?

File: wemake_python_styleguide/visitors/tokenize/comments.py (hunk 3 above)
Comment (edytagarbarz) [reply wemake_python_styleguide/visitors/tokenize/comments.py]: Changed to `noqa_code`

File: wemake_python_styleguide/visitors/tokenize/comments.py (hunk 1 above)
Comment (edytagarbarz) [reply wemake_python_styleguide/visitors/tokenize/comments.py]: Done

**Comment by edytagarbarz** [comment]
> Hi @sobolevn I need your help here with logistic. I added this new violation under MagicCommentVisitor but doing so I exceeded allowed number of methods. I thought about moving noqa comments checks to a separate visitor but wanted to hear your thoughts first. Thanks!

I still have a problem with this. Thanks!

**Comment by sobolevn** [comment]
Lets move it to a new one, please!

**Comment by edytagarbarz** [comment]
Done!

**Comment by KumarGanesha1996** [comment]
> * [] I have added my changes to the `CHANGELOG.md`

please fix checkbox in description

**Review by sobolevn — APPROVED** [review]

**Pull request merged**